theories/Core/gen_type_classes.ml

```diff
@@ -0,0 +1,60 @@
+open Format
+
+let max_interfaces = 20
```

Comment:
Why 20? 

Did you measure the slowdown induced by generating that many instances and classes?